controllers/gateway_controller.go

```diff
@@ -491,15 +474,13 @@ func UpdateGWListenerStatus(ctx context.Context, k8sclient client.Client, gw *ga
 	glog.V(6).Infof("After update, the snapshot of listener status %v", gw.Status.Listeners)
 
 	if err := k8sclient.Status().Patch(ctx, gw, client.MergeFrom(gwOld)); err != nil {
-		glog.V(2).Infof("Failed to update gateway listener err: %v, status: %v", err, gw.Status.Listeners)
-		return errors.Wrapf(err, "failed to update gateway status")
+		return errors.Wrapf(err, "failed to update gateway listener err: %v, status: %v", err, gw.Status.Listeners)
 	}
 
 	if hasValidListener {
 		return nil
 	} else {
-		glog.V(2).Infof("no valid listeners for %v\n", gw.Name)
-		return errors.New("invalid listeners")
+		return fmt.Errorf("no valid listeners for %v\n", gw.Name)
```

Comment:
nit: Is a new line still required here?

Reply:
no, missed \n, will fix other occurrences too